UM-USElab/gradecraft-development
# Prevent a learning objective level from being deleted if it has been used to assess students

### Status
**READY**

### Description
* It should not be possible to delete a learning objective level if it has been used to assess a student (i.e. if an instance of LearningObjectiveObservedOutcome exists with a link to the level). Currently, any learning objective level can be deleted, which can lead to a "hanging" outcome where instances of LearningObjectiveObservedOutcome reference levels that do not exist.
* Now, if the level has been used to assess students, a message is displayed asking instructors to reassess students with a different level before deleting it.

### Migrations
NO

### Steps to Test or Reproduce
1. As an instructor, create an assignment with a linked learning objective, create a level for the learning objective. Grade a student and award the learning objective to the student with the level that was created.
2. Edit the learning objective and delete the learning objective level created. A message is shown that asks to reassess students with a different level as the level being deleted is currently being used to assess students.

### Impacted Areas in Application
* Learning objective levels controller (i.e. controllers/api/learning_objectives/levels_controller.rb)

======================
Closes #3938
